server/src/main/java/com/linecorp/centraldogma/server/internal/storage/ScheduledPurgeService.java

```diff
@@ -36,64 +39,131 @@
 import com.google.common.util.concurrent.ListenableScheduledFuture;
 import com.google.common.util.concurrent.ListeningScheduledExecutorService;
 import com.google.common.util.concurrent.MoreExecutors;
+import com.spotify.futures.CompletableFutures;
 
 import com.linecorp.centraldogma.common.Author;
 import com.linecorp.centraldogma.server.command.Command;
 import com.linecorp.centraldogma.server.command.CommandExecutor;
+import com.linecorp.centraldogma.server.metadata.MetadataService;
+import com.linecorp.centraldogma.server.metadata.UserAndTimestamp;
 import com.linecorp.centraldogma.server.storage.project.ProjectManager;
 
 import io.netty.util.concurrent.DefaultThreadFactory;
 
 /**
  * A service class for pursing projects and repositories that were removed before.
  */
-public class StoragePurgeService {
+public class ScheduledPurgeService {
```

Comment:
I was also thinking about `Plugin` too. Because it has a nice start/stop code as you said. 
I hesitated to implement this code as `Plugin` because I don't have a strong conviction that this feature is suitable as `Plugin`.
Thanks for nice advice and I try to make this a `Plugin`. 😊

Reply:
Added [ScheduledPurgingServicePlugin](https://github.com/line/centraldogma/pull/416/files#diff-1522f6e7ffae1368385b728f4d695d0a)